core/primitives/src/runtime/config_store.rs

```diff
@@ -0,0 +1,103 @@
+use include_dir::{include_dir, Dir};
+
+use crate::runtime::config::RuntimeConfig;
+use crate::types::Gas;
+use crate::types::ProtocolVersion;
+use std::collections::BTreeMap;
+use std::iter::FromIterator;
+use std::ops::Bound;
+use std::sync::Arc;
+
+/// Stores runtime config for each protocol version where it was updated.
+#[derive(Debug)]
+pub struct RuntimeConfigStore {
+    /// Maps protocol version to the config with possibly modified `max_gas_burnt_view` limit.
+    store: BTreeMap<ProtocolVersion, Arc<RuntimeConfig>>,
+}
+
+impl RuntimeConfigStore {
+    /// Constructs a new store.
+    ///
+    /// If `max_gas_burnt_view` is provided, the property in wasm limit
+    /// configuration will be adjusted to given value.
+    pub fn new(max_gas_burnt_view: Option<Gas>) -> Self {
```

Comment:
What is special about `max_gas_burnt_view` that it is not stored as part of the file?